docs/source/contributing/layout.rst

```diff
@@ -0,0 +1,126 @@
+
+onbuild/
+    This directory contains the Dockerfile for building a JupyterHub image. Specifically, it would include the configuration file(`JupyterHub_config.py`) in the image build.
+
+share/jupyterHub/
+    This directory contains JupyterHub data and static files.
+
+singleuser/
+    This directory contains the Dockerfile for building a JupyterHub image for a single user. You can check out the `README` in the directory for more details.
+
+testing/
+    This directory contains a sample JupyterHub configuration file for testing.
+
+Executable files
+================
+
+setup.py
+    The packaging file for the JupyterHub codebase. It is used in building the JupyterHub distribution package. It contains information on the name, version, data files, package files, python version requirement, entry points, and the dependencies of the JupyterHub codebase. 
+
+Dockerfile
+    The Dockerfile containing the base image for running JupyterHub.
+
+bower-lite
+    Another components file used in packaging JupyterHub. It stages the frontend dependencies of the JupyterHub repository from the node_modules directory into the components sub-directory. Check the static directory in the `share/jupyterHub` directory to find the components sub-directory.
+
+package.json
+    This file contains information on JupyterHub nodejs dependencies.
+
+pytest.ini
+    A configuration file for customizing Pytest behaviors to suit the usage of Pytest in the JupyterHub project. Pytest is a python tool for running tests.
+
+pyproject.toml
+    A requirement for using setuptools(a build backend) in building the JupyterHub distribution package.
+
+MANIFEST.in
+    A file for instructing setuptools on the files to add or remove when building the JupyterHub distribution package (sdist).
+
+requirements.txt
+    This file lists the dependencies of the JupyterHub codebase.
+
+dev-requirements.txt
+    This file lists additional dependencies of the JupyterHub codebase.
+
+
+Informational Files
+===================
+
+README
+    Read the README! Read it first.
+
+    This file contains the minimum documentation to help you get started with the JupyterHub project. It covers installations, some notes on contribution, and a list of help and resources. It points to the main documentation of the project.
+
+CODE_OF_CONDUCT.md
+    This file links to Project Jupyter's code of conduct.
+
+CONTRIBUTING.md
+    This file points to the JupyterHub contribution guidelines and other resources such as setting up a developmental install to help new contributors get started making contribution to the project.
+
+COPYING.md
+    This file contains the JupyterHub license information. It explains the permission you have using source code from the repository.
+
+RELEASE.md
+    This file provides information on making releases of the JupyterHub project.
+
+SECURITY.md
+    This file provides information on how to report an identified security vulnerability in the JupyterHub project.
+
+
+dotfiles
```

Comment:
dotfiles aren't a separate category - these are mostly the same as what's called 'executable files' right now. Some of them just happen to start with '.'